appender.go

```diff
@@ -341,6 +340,7 @@ func (a *Appender) flush(ctx context.Context, bulkIndexer *BulkIndexer) error {
 		a.addCount(int64(flushed), &a.bytesUncompressedTotal, a.metrics.bytesUncompressedTotal)
 	}
 	if err != nil {
+		defer a.addUpDownCount(-int64(n), &a.docsActive, a.metrics.docsActive)
```

Comment:
Do we need the `defer` now? The actual flushing is done at this point. Same below